src/main/webapp/nearmeScript.js

```diff
@@ -23,26 +29,122 @@ function initializeMap() {
-  console.log(map);
-  console.log(currentLocation);
-  var request = {
-    location: currentLocation,
-    radius: '500',
-    type: ['restaurant']
-  };
-  service = new google.maps.places.PlacesService(map);
-  service.nearbySearch(request, callback);
-  // output response of API call to console
-  function callback(results, status) {
-    if (status == google.maps.places.PlacesServiceStatus.OK) {
-      console.log(results);
+  eventsDivElement = document.getElementById('nearbyEvents');
+  eventsDivElement.innerHTML = '';
+  var geocoder = new google.maps.Geocoder();
+  // This is the circle within which we search for events.
+  var locationCircle = new google.maps.Circle({ 
+    map: map,
+    center: currentLocation,
+    radius: 2000 
+  });
+  
+  fetch('/events')
+  .then(response => response.json())
+  .then(events => {
+    // When user is logged in, get all public events and events user is attending.
+    if (localStorage.getItem('loginStatus').localeCompare('true') == 0) {
+      var userId = localStorage.getItem('userId');
+      for (var i = 0; i < events.length; i++) {
+        var currentEvent = events[i];
+        isNearby(geocoder, currentEvent, locationCircle, userId);
+      }
     }
+    // When user is not logged in, get only public events.
     else {
-      alert(status);
+      for (var i = 0; i < events.length; i++) {
+        var currentEvent = events[i];
+        isNearby(geocoder, currentEvent, locationCircle, '');
+      }
     }
-  }
-  // link this function to nearme page
-  // test
+  });
 }
+
+/** Checks to see if an event is nearby */
+function isNearby(geocoder, event, locationCircle, userId) {
+  geocoder.geocode( {'placeId' : event.placeId}, function(results, status) {
+    if (status !== google.maps.GeocoderStatus.OK) {
+      alert('Geocode was not successful for the following reason: ' + status);
+      return;
+    }
+    eventLatLng = results[0].geometry.location;
+    var isNearby = locationCircle.getBounds().contains(eventLatLng)
+    if (isNearby) {
+      if (userId) {
+        if (event.rsvpAttendees.includes(userId) || 
+          event.privacy == 'public') { 
+          eventElement = createNearMeEvent(event);
+          eventElement.addEventListener('click', () => {
+            sessionStorage.setItem('currentLocationId', event.placeId);
+            window.location.href = 'map.html';
+          });
+           eventsDivElement.appendChild(eventElement);
+        }
+      }
+      else {
+        if (event.privacy == 'public'){
+          eventElement = createNearMeEvent(event);
+          eventElement.addEventListener('click', () => {
+            sessionStorage.setItem('currentLocationId', event.placeId);
+            window.location.href = 'map.html';
+          });
+          eventsDivElement.appendChild(eventElement);
+        }
+      } 
+      
+    }
+  });
+}
+
+/** Creates a display element for an event. */
+function createNearMeEvent(event) {
```

Comment:
NIT: It might be easier / less lines of code to just call the create event function in the map page?

Reply:
Oh that'll actually be better. Ill do that instead